app/controllers/assessments_controller.rb

```diff
@@ -290,20 +291,26 @@ def importAssessment
 
   def create
     @assessment = @course.assessments.new(new_assessment_params)
-
     if @assessment.name.blank?
-      # Validate the name
-      ass_name = @assessment.display_name.downcase.gsub(/[^a-z0-9]/, "")
-
-      if ass_name.blank?
+      # Validate the name, very similar to valid Ruby identifiers, but also allowing hyphens
+      # We just want to prevent file traversal attacks here, and stop names that break routing
+      # first regex - try to sanitize input, allow special characters in display name but not name
+      # if the sanitized doesn't match the required identifier structure, then we reject
+      sanitized_display_name = @assessment.display_name.gsub(/[!@#$%^&*(),.?":{}|<>\s]/, "")
+      Rails.logger.debug(sanitized_display_name)
```

Comment:
Nit: remove this debug line, and the todo comment below.

Also, it does not let me create an assessment with name `_4utol4b-_4smt` (as per docs)

Reply:
It allows me to create `.autolab.asmt` which is supposed to be invalid

Reply:
the behavior here has been changed to rather than just try to substitute out bad characters, take out a portion of the display name that can be used as a valid name. so now `.autolab.asmt` will become `autolab`. Does this sound reasonable?

Reply:
Makes sense